### ISSUE TITLE
Hangs if network is unreachable

### ISSUE THREAD

**Issue by lessthanoptimal** [axet/wget#22]
As of this morning this address is unreachable (might change in the future) and java wget just hangs.  Everything else fails immediately when I connect to it.

Using this version: compile 'com.github.axet:wget:1.3.1' on Ubuntu 16.04

```
		DownloadInfo info = new DownloadInfo(new URL("http://heanet.dl.sourceforge.net/project/deepboof/networks/v1/likevgg_cifar10.zip"));
		info.extract();
		WGet w = new WGet(info, new File("junk"));
		w.download();
```

**Comment by axet** [comment]
Is it hangs or auto retries for your comfort?

**Comment by lessthanoptimal** [comment]
Is there a difference between auto-retrying forever/at least 10 minutes and freezing?  I did look for a way to set a timeout in the WGet class without success.  Switched it to using:

public void download(AtomicBoolean stop, Runnable notify)

to see if it would maybe notify me that something was up and that wasn't the case.

Ended up coding up a customized routine of my own to download the file as a work around that's better suited for my specific application.
